### PR TITLE
Add nodes caching for cells

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR changes the way how the DOM elements are rendered. Previously, the engine created cells anew for each rendering cycle (slow render) and filled them with content by calling the renderer function assigned to it. After the changes, all cells are created once, cached, and reused for each new rendering cycle. Only the content of the cells is updated. Thanks to that, the table (depending on the viewport size) renders faster. From my testing and observations, depending on the test case (navigation, scrolling, etc.), it speeds up to 40% faster.

For `renderAllRows: true` and/or `renderAllColumns: true` (no virtualization) there are no changes in performance.

_[skip changelog]_

### How has this been tested?

I tested the changes locally and updated all tests according to the new changes.

### Types of changes

- [x] New feature or improvement (non-breaking change which adds functionality)

### Related issue(s):
1. https://github.com/handsontable/dev-handsontable/issues/779
2. fixes https://github.com/handsontable/dev-handsontable/issues/2099

### Affected project(s):
- [x] `handsontable`
- [x] `@handsontable/angular`
- [x] `@handsontable/react`
- [x] `@handsontable/react-wrapper`
- [x] `@handsontable/vue`
- [x] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)

### CODE REVIEW

File: handsontable/src/3rdparty/walkontable/src/renderer/cells.js [hunk 1]
```diff
@@ -91,9 +89,14 @@ export default class CellsRenderer extends BaseRenderer {
       for (let visibleColumnIndex = 0; visibleColumnIndex < columnsToRender; visibleColumnIndex++) {
         orderView.render();
 
-        const TD = orderView.getCurrentNode();
         const sourceColumnIndex = this.table.renderedColumnToSource(visibleColumnIndex);
 
+        // if (!orderView.hasStaleContent(sourceColumnIndex)) {
+        //   continue;
+        // }
+
```
Comment (jansiegel) [inline handsontable/src/3rdparty/walkontable/src/renderer/cells.js:97]: Probably a leftover?

File: handsontable/src/3rdparty/walkontable/src/utils/nodesPool.js [hunk 2]
```diff
@@ -27,9 +33,23 @@ export default class NodesPool {
...
+    const hasColumnIndex = typeof columnIndex === 'number';
+    const key = hasColumnIndex ? `${rowIndex}x${columnIndex}` : rowIndex.toString();
+
+    if (this.pool.has(key)) {
+      return this.pool.get(key);
+    }
+
+    const node = this.rootDocument.createElement(this.nodeType);
+
+    // node.dataset.id = key; // Uncomment for debug purposes
```
Comment (jansiegel) [inline handsontable/src/3rdparty/walkontable/src/utils/nodesPool.js:50]: Not sure if this is meant to stay here for future use, or was just a temporary thing?

**Review by jansiegel — CHANGES_REQUESTED** [review]

File: handsontable/src/3rdparty/walkontable/src/utils/nodesPool.js (hunk 2 above)
Comment (budnix) [reply handsontable/src/3rdparty/walkontable/src/utils/nodesPool.js]: done https://github.com/handsontable/handsontable/pull/11264/commits/c65364b9b787d789b33e8ebd800b7cdb0efbaa00

File: handsontable/src/3rdparty/walkontable/src/renderer/cells.js (hunk 1 above)
Comment (budnix) [reply handsontable/src/3rdparty/walkontable/src/renderer/cells.js]: Done https://github.com/handsontable/handsontable/pull/11264/commits/c65364b9b787d789b33e8ebd800b7cdb0efbaa00

**Review by jansiegel — APPROVED** [review]

**Pull request merged**